test/e2e/autoscale_test.go

```diff
@@ -314,19 +328,24 @@ func assertAutoscaleUpToNumPods(ctx *testContext, numPods int32) {
 
 	assertNumberOfPodsEvery(2*time.Second, ctx, errChan, stopChan, minPods, maxPods)
 
-	if err := generateTraffic(ctx, int(numPods*10), 30*time.Second); err != nil {
-		ctx.t.Fatalf("Error during scale up: %v", err)
-	}
+	go generateTraffic(ctx, int(numPods*10), timeout, errChan, stopChan)
 
-	if err := assertNumberOfPods(ctx, minPods, maxPods); err != nil {
-		errChan <- err
-	}
+	done := make(chan bool)
+	go func() {
+		for {
+			if err := assertNumberOfPods(ctx, numPods, maxPods); err == nil {
+				done <- true
+			}
+		}
+	}()
```

Comment:
This looks like a busy loop to me. It should have a wait period in between assert calls I think.

However, `assertNumberOfPodsEvery` seems to have a very similar purpose. WDYT about inlining that here for some extra clarity. I'd envision that to look like this:

```golang
done := make(chan struct{})
timer := time.Tick(interval)
go func() {
	for {
		select {
		case <-stopChan:
			return
		case <-timer:
			if err := assertNumberOfPods(ctx, minPods, maxPods); err != nil {
				errChan <- err
			}
			if err := assertNumberOfPods(ctx, numPods, maxPods); err == nil {
				close(done)
			}
		}
	}
}()
```

(Mostly a copy of assertNumberOfPodsEvery but added the extra if you have in this loop).

Also note: You can simply close a channel to send a done signal. Bonus points for making the channel of type `struct{}` so it doesn't allocate unnecessary space.

Reply:
thanks for the pointer.
still learning the nuances of golang.

much better now without assertNumberOfPodsEvery